module-2/backend/week_three.md

```diff
@@ -10,14 +10,44 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 1. What is the entry at the command line to create a new rails app?
+  rails new app_name
+  
 2. What do Models generally inherit from in rails?
+  ActiveRecord::Base
+  
 3. What do Controllers generally inherit from in a rails project?
-4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+  ApplicationController
+  
+4. How would I create a route if I wanted to see a specific horse in my routes file assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+  resources :horses, only: [:show]
+  
 5. What rake task is useful when looking at routes, and what information does it give you?
+  rake routes - It will give you what paths are available to you, the uri pattern and the HTTP verb. 
+  
 6. What is an example of a route helper? When would you use them?
+  new_example_path - It would be used when wanting to access the page/route to create a new "example"
+
 7. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
+    `_path` will return a relative URI like "/doohickeys"
+    `_url` will return an absolute URI like "http://www.example.com/doohickeys"
+
 8. What are strong params and why are the necessary?
+  it controls what user can present to the application and it is done to protect the model 
+
 9. What role does `form_for` play in helping us create our forms?
+  it streamlines the process of creating forms and even allowing for it be reused in other views. It will also generate names according to the used "view"
+
 10. How does `form_for` know where to submit the user's input?
+  because of the passed instance variable and the view template that it lives in
```

Comment:
it checks if that instance variable is a new record or if it has been in the database - then it know it it needs to send to form to post or put